### ISSUE TITLE
SMAR-17 ⁃ Route Planning

### ISSUE THREAD

**Issue by thatdoogieguy** [thatdoogieguy/SST#129]
Route planning will be an important feature for the Smart Sensor Solution. The idea behind route planning is to create efficiencies in bin collection through smart use of data. This builds on the concept of Just In Time Waste Management.

Route Planning will utilise the sensor data, both current and historical to indicate which bins in a defined zone need to be emptied and plot out a route accordingly. To take this a step further, the feature should be able to be automated and drivers assigned to allow the system to function with minor intervention / review.

Key features of the Route Management System will include:

* Route Groups & Auto Grouping (can be zones, a group of zones, or a subset of zones)
* Route Options

* - Estimated Time at Site / Bin Collection Time per bin.
* - Route frequency / Start and Finish times.
* - Calculate Route time and distance

* Collection thresholds

* - Data based collection
* - Min and Max collection levels
* - Days since last collected parameter
* - All Bins Override

* Driver Roster Management / Allocation
* Vehicle / Fleet Management / Allocation
* Route Plotting

* - Quickest
* - Shortest

* Manual Route Tweaking
* Automation
* Route Efficiency Reporting

In a nutshell, the Manager / User should be able to setup a group of bins into a route group, allocate a driver and a vehicle, and then set parameters to generate a route for the driver.

This should be delivered to the driver through the App and Website so they can pull up and review their route.

This process should be able to be automated on a regular basis, recurring at set intervals allowing for the routes to run on autopilot.

Routes should then be able to be reviewed for efficiency and overall time taken to complete the route.
Routes
